extensions/src/AWSSDK.Extensions.CborProtocol/Internal/Transform/CborSimpleTypeUnmarshaller.cs

```diff
@@ -0,0 +1,559 @@
+﻿/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+using System;
+using System.Collections.Generic;
+using System.Formats.Cbor;
+using System.IO;
+using Amazon;
+using Amazon.Runtime;
+using Amazon.Runtime.Internal.Util;
+using Amazon.Util;
+
+namespace AWSSDK.Extensions.CborProtocol.Internal.Transform
+{
+    static class CborSimpleTypeUnmarshaller<T>
+    {
+        public static T Unmarshall(CborUnmarshallerContext context)
+        {
+            var reader = context.Reader;
+            object value;
+
+            if (
+                (
+                    typeof(T) == typeof(int?) ||
```

Comment:
nit: Do we need all the type checks? If `reader.PeekState()` is `null` then shouldn't we always call `reader.ReadNull()` and set `value` to `null`?

Reply:
Ok